dso_pkg/dso/policy_optimizer/policy_optimizer.py

```
@@ -253,7 +257,34 @@ def train_step(self,
+            # Add entropy regularization if available
+            if hasattr(self.policy, 'entropy_weight'):
+                entropy = self.policy.entropy(sampled_batch.obs, sampled_batch.actions)
+                entropy_loss = -self.policy.entropy_weight * tf.reduce_mean(entropy)
+                total_loss = pg_loss + entropy_loss
+            else:
+                total_loss = pg_loss
+        
+        # Compute gradients and apply
+        gradients = tape.gradient(total_loss, self.policy.controller.trainable_variables)
+        self.optimizer_instance.apply_gradients(
+            zip(gradients, self.policy.controller.trainable_variables)
```

Ruff (0.12.2)

`zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)